chain-impl-mockchain/src/ledger/ledger.rs

```diff
@@ -1015,3 +1014,1188 @@ fn input_multi_account_verify<'a>(
     ledger = new_ledger;
     Ok(ledger)
 }
+
+#[cfg(test)]
+mod tests {
+
+    use super::*;
+    use crate::{
+        account::{Identifier, SpendingCounter},
+        accounting::account::account_state::AccountState,
+        key::Hash,
+        multisig,
+        testing::{
+            address::ArbitraryAddressDataValueVec,
+            builders::{witness_builder, TransactionBuilder},
+            data::{AddressData, AddressDataValue},
+            ledger::{self as ledger_mock, ConfigBuilder},
+            TestGen,
+        },
+        transaction::Witness,
+    };
+    use chain_addr::Discrimination;
+    use quickcheck::{Arbitrary, Gen, TestResult};
+    use quickcheck_macros::quickcheck;
+    use std::iter;
+
+    impl Arbitrary for LedgerStaticParameters {
+        fn arbitrary<G: Gen>(g: &mut G) -> Self {
+            LedgerStaticParameters {
+                block0_initial_hash: Arbitrary::arbitrary(g),
+                block0_start_time: Arbitrary::arbitrary(g),
+                discrimination: Arbitrary::arbitrary(g),
+                kes_update_speed: Arbitrary::arbitrary(g),
+            }
+        }
+    }
+
+    impl Arbitrary for LedgerParameters {
+        fn arbitrary<G: Gen>(g: &mut G) -> Self {
+            LedgerParameters {
+                fees: Arbitrary::arbitrary(g),
+            }
+        }
+    }
+
+    #[quickcheck]
+    pub fn match_identifier_witness_prop_test(
+        id: AccountIdentifier,
+        witness: Witness,
+    ) -> TestResult {
+        let result = super::match_identifier_witness(&id, &witness);
+        match (witness.clone(), result) {
```

Comment:
`clone` should not be necessary, but if it saves some syntax pain, it's OK for test code.